doc/manual/source/glossary.md

```diff
@@ -31,9 +31,22 @@
 
   The industry term for storage and retrieval systems using [content addressing](#gloss-content-address). A Nix store also has [input addressing](#gloss-input-addressed-store-object), and metadata.
 
+- [derivation]{#gloss-derivation}
+
+  A derivation can be thought of as a [pure function](https://en.wikipedia.org/wiki/Pure_function) that produces new [store objects][store object] from existing store objects.
```

Comment:
This is not true because a function maps many possible arguments to results (many possible inputs) but a derivation has a specific set of inputs.

A derivation that depends on the outputs of another derivation is closer to being a function, because we don't know what those outputs are at the time of creating the derivation, but even this needs more explaining.

I think it would be better to say a derivation is a function application than a function. since it is applying a function-in-our-heads to some inputs.